engine/src/main/java/org/terasology/rendering/opengl/GLSLShader.java

```diff
@@ -322,85 +328,68 @@ private void compileShader(int type, Set<ShaderProgramFeature> features) {
             shader.append(shaderProgramBase.getVertexProgram());
         }
 
+        return shader.toString();
```

Comment:
I know you didn't touch that bit, but couldn't the two if blocks above be conflated into one?